pace-util/pace/util/zarr_monitor.py

```diff
@@ -264,16 +273,24 @@ def append(self, quantity):
 
     def _get_attrs(self, quantity):
         return {
-            "_ARRAY_DIMENSIONS": list(self._PREPEND_DIMS + quantity.dims),
+            "_ARRAY_DIMENSIONS": self._get_quantity_dims(quantity),
             **quantity.attrs,
         }
 
-    def _ensure_compatible_attrs(self, new_quantity):
-        new_attrs = self._get_attrs(new_quantity)
-        if dict(self.array.attrs) != new_attrs:
+    def _get_quantity_dims(self, quantity):
+        return list(self._PREPEND_DIMS + quantity.dims)
+
+    def _transpose(self, quantity):
+        transposed = quantity.transpose(self.array_dims[2:])
+        transposed.update_attrs(dissoc(quantity.attrs, "units"))
+        return transposed
```

Comment:
I would suggest deleting this helper, and instead modifying `quantity.transpose` to propagate attrs.